### PR TITLE
Switch to ocaml-lsp-server

### PR DESCRIPTION
**Pull request by rusty-key**
`ocamlmerlin-lsp` was moved to a dedicated repo and renamed to `ocaml-lsp-server`, binary was renamed to `ocamllsp`. This PR switches extension to the new version.

### CODE REVIEW

**Comment by prometheansacrifice** [comment]
Not sure why the CI failed - @andreypopp Can you please re-run the pipeline?

**Comment by andreypopp** [comment]
done: https://dev.azure.com/arrowresearch/vscode-merlin/_build/results?buildId=27&view=results

File: azure-pipelines.yaml [hunk 1]
```diff
@@ -42,7 +42,7 @@ steps:
 
 - bash: |
     echo ">>> Compile vscode-test"
-    npm i && npm run compile
+    npm i && npm run build
```
Comment (rusty-key) [inline azure-pipelines.yaml]: This is not related. Just noticed that there is an error in the pipeline

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -20,8 +20,11 @@ steps:
     versionSpec: '8.x'
   displayName: 'Install Node.js'
 
-- script: 'npm install -g esy@0.5.0 --unsafe-perm'
-  displayName: 'npm install -g esy'
+# Use nightly build while 0.6.* is not released
+# - script: 'npm install -g esy@0.5.0 --unsafe-perm'
+#   displayName: 'npm install -g esy'
+- script: 'npm install -g @esy-nightly/esy --unsafe-perm'
+  displayName: 'npm install -g @esy-nightly/esy'
```
Comment (rusty-key) [inline azure-pipelines.yaml]: Realized that we need `esy@0.6` in order to build `ocamllsp`. But CI still reports lack of `diagnostic` (expected 1, actual 0):
https://dev.azure.com/arrowresearch/vscode-merlin/_build/results?buildId=40&view=logs&j=291b3e6c-60a9-5f30-6b17-b27c7388dcd0&t=7c007d9f-47e6-568b-fd3e-7edb68926b71&l=785

Not sure what else can I do. My theory is that `ocamllsp` is not getting compiled for some reason but it is not reported. @andreypopp @prometheansacrifice, any ideas how I can check/fix that?

**Comment by rusty-key** [comment]
Finally realized that it's a problem with the test setup, fixed now.
